aws/checksum.go

```diff
@@ -0,0 +1,39 @@
+package aws
+
+// RequestChecksumCalculation controls request checksum calculation workflow
+type RequestChecksumCalculation string
+
+const (
+	// RequestChecksumCalculationWhenSupported indicates request checksum should be calculated if modeled
+	RequestChecksumCalculationWhenSupported RequestChecksumCalculation = "whenSupported"
+
+	// RequestChecksumCalculationWhenRequired indicates request checksum should be calculated
+	// if modeled and user set an algorithm
+	RequestChecksumCalculationWhenRequired = "whenRequired"
+)
+
+// ResponseChecksumValidation controls response checksum validation workflow
+type ResponseChecksumValidation string
+
+const (
+	// ResponseChecksumValidationWhenSupported indicates response checksum should be validated if modeled
+	ResponseChecksumValidationWhenSupported ResponseChecksumValidation = "whenSupported"
+
+	// ResponseChecksumValidationWhenRequired indicates response checksum should be validated if modeled
+	// and user enable that in vlidation mode cfg
+	ResponseChecksumValidationWhenRequired = "whenRequired"
+)
+
+// RequireChecksum indicates if a checksum needs calculated/validated for a request/response
+type RequireChecksum string
+
+const (
```

Comment:
A few comments though I think they'll be moot in the end:
* this should definitely be an internal type
* this is not an idiomatic way to represent a ternary boolean - ternary bool should almost _never_ be necessary in Go, but if it absolutely is, you'd just use `*bool`